maven-core/src/main/java/org/apache/maven/internal/aether/PropertyContributorExtender.java

```diff
@@ -50,13 +54,53 @@ class PropertyContributorExtender implements MavenExecutionRequestExtender {
     public void extend(MavenExecutionRequest mavenExecutionRequest) {
         Map<String, PropertyContributor> effectivePropertyContributors = lookup.lookupMap(PropertyContributor.class);
         if (!effectivePropertyContributors.isEmpty()) {
-            HashMap<String, String> userPropertiesMap = new HashMap<>((Map) mavenExecutionRequest.getUserProperties());
+            final Version mavenVersion = lookup.lookup(VersionParser.class)
+                    .parseVersion(lookup.lookup(RuntimeInformation.class).getMavenVersion());
+            final Map<String, String> systemPropertiesMap =
+                    Map.copyOf((Map) mavenExecutionRequest.getSystemProperties());
+            final Map<String, String> userPropertiesMap = Map.copyOf((Map) mavenExecutionRequest.getUserProperties());
+            final ProtoSession protoSession = new ProtoSession() {
+                @Override
+                public Version getMavenVersion() {
+                    return mavenVersion;
+                }
+
+                @Override
+                public Map<String, String> getUserProperties() {
+                    return userPropertiesMap;
+                }
+
+                @Override
+                public Map<String, String> getSystemProperties() {
+                    return systemPropertiesMap;
+                }
+
+                @Override
+                public Path getTopDirectory() {
+                    return mavenExecutionRequest.getTopDirectory();
+                }
+
+                @Override
+                public Path getRootDirectory() {
+                    return mavenExecutionRequest.getRootDirectory();
+                }
+            };
+
+            final Properties newProperties = new Properties();
+
             for (PropertyContributor contributor : effectivePropertyContributors.values()) {
-                contributor.contribute(userPropertiesMap);
+                Map<String, String> contribution = contributor.contribute(protoSession);
```

Comment:
A given contributor won't see properties added by previous contributors, is that what we want ?